src/inmanta/data/dataview.py

```diff
@@ -966,10 +1000,13 @@ def get_base_query(self) -> SimpleQueryBuilder:
                 -- Get all resource action in the given environment for the given resource_id
                 WITH actions AS (
                     SELECT  ra.*
-                    FROM {Resource.table_name()} AS r INNER JOIN resourceaction_resource AS rr ON (
+                    FROM resource_set_configuration_model AS rscm
+                    INNER JOIN {Resource.table_name()} AS r
+                        ON rscm.environment=r.environment AND rscm.resource_set_id=r.resource_set_id
+                    INNER JOIN resourceaction_resource AS rr ON (
                                                           r.environment=rr.environment
                                                           AND r.resource_id=rr.resource_id
-                                                          AND r.model=rr.resource_version
+                                                          AND rscm.model=rr.resource_version
```

Comment:
I think we can drop all lot of joins here? 

We have the resource_id and use it on the  `resourceaction_resource` that we join with `resourceaction` 
That is all we need, right?

Reply:
I think we also need to join on `resourceaction_resource.resource_version`, no?